### PR TITLE
Create a crate for querying WMI

### PR DESCRIPTION
**Pull request by panhania**

### CODE REVIEW

File: crates/wmi/src/windows.rs [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+
+        let mut row = std::collections::HashMap::new();
+
+        loop {
+            let mut raw_name = std::mem::MaybeUninit::uninit();
+            let mut raw_value = std::mem::MaybeUninit::uninit();
+
+            // SAFETY: We advance the iterator [1] without any extra flags. Note
+            // that we already called `BeginEnumeration`. The call should call
+            // `VariantInit` on `raw_value` [2] so we should not do it upfront.
```
Comment (s-westphal) [inline crates/wmi/src/windows.rs:170]: The link for [2] seems to be missing

File: crates/wmi/src/windows/com.rs [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        _: &'com InitGuard,
+        ptr: *mut super::ffi::IEnumWbemClassObject,
+    ) -> EnumWbemClassObject<'com> {
+        EnumWbemClassObject {
+            ptr,
+            com: std::marker::PhantomData,
+        }
+    }
+
+    /// Returns reference the underlying COM object.
```
Comment (s-westphal) [inline crates/wmi/src/windows/com.rs:260]: Nit: ...reference **to** the...  (3 more times)

File: crates/wmi/src/windows/com.rs [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+            )
+        };
+
+        match status {
+            windows_sys::Win32::System::Wmi::WBEM_S_NO_ERROR => {
+                // SAFETY: This branch is reached only if the call succeeded and
+                // the returned count is equal to the requested count. We can
+                // thus assume that `count` is initialized and just as a sanity
+                // check we verify that it matches 1 (since we only requested
+                // a single result.
```
Comment (s-westphal) [inline crates/wmi/src/windows/com.rs:322]: Nit: add missing ")"

**Review by s-westphal — APPROVED** [review]

File: crates/wmi/src/windows.rs (hunk 1 above)
Comment (panhania) [reply crates/wmi/src/windows.rs]: Rephrased and removed `[2]` as it refers to content from `[1]` anyway.

File: crates/wmi/src/windows/com.rs (hunk 2 above)
Comment (panhania) [reply crates/wmi/src/windows/com.rs]: Done.

File: crates/wmi/src/windows/com.rs (hunk 3 above)
Comment (panhania) [reply crates/wmi/src/windows/com.rs]: Done.

**Pull request merged**